### PR TITLE
Add Debian support

### PR DESCRIPTION
**Pull request by ekohl**
This allows you to provision a Foreman 1.14 install on Debian. There are
still many corners that problably don't support it and Katello boxes are
also generated but it's a start.

### CODE REVIEW

**Comment by mmoll** [comment]
I don't use forklift (yet?), but I'd say if this works also for other people: :shipit: 🚀

**Comment by ekohl** [comment]
Rebased.

**Comment by ehelms** [comment]
@ekohl needs another rebase

**Comment by ekohl** [comment]
Rebased

**Comment by evgeni** [comment]
Also, I was running with the following changes:

```patch
diff --git a/config/versions.yaml b/config/versions.yaml
index e3b2a2d..9d255c3 100644
--- a/config/versions.yaml
+++ b/config/versions.yaml
@@ -31,6 +31,7 @@ installers:
     puppet: 4
     boxes:
       - 'centos7'
+      - 'debian8'
 
   - foreman: 'nightly'
     katello: 'nightly'
diff --git a/playbooks/roles/foreman_repositories/tasks/main.yml b/playbooks/roles/foreman_repositories/tasks/main.yml
index f3cf210..38d8f42 100644
--- a/playbooks/roles/foreman_repositories/tasks/main.yml
+++ b/playbooks/roles/foreman_repositories/tasks/main.yml
@@ -12,3 +12,4 @@
   yum:
     name: foreman-release-scl
     state: latest
+  when: ansible_os_family == 'RedHat'
```

File: playbooks/roles/foreman_installer/tasks/locales.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+- name: 'Install locales'
+  apt:
+    name: locales
+    state: latest
+
+- name: 'Ensure en_US.UTF-8 locale'
+  lineinfile:
```
Comment (ekohl) [inline playbooks/roles/foreman_installer/tasks/locales.yml]: Because I did not know it existed, thanks!

**Comment by evgeni** [comment]
FTR, it fails for me with:

```
[ERROR 2017-06-12 14:24:49 main]  /Stage[main]/Postgresql::Server::Initdb/File[/var/lib/postgresql/9.4/main]/ensure: change from absent to directory failed: Cannot create /var/lib/postgresql/9.4/main; parent directory /var/lib/postgresql/9.4 does not exist
```

But the real issue is that the PostgreSQL packages did not set up properly neither /var/lib/postgresql/9.4 nor /etc/postgresql/9.4 due to locale problems.

**Comment by ekohl** [comment]
I don't have a proper environment right now to test this, but I fixed some code up to have names (makes the tests pass as well) and use locale_gen. Will give it a spin when I get back to my dev env.

File: playbooks/roles/foreman_repositories/tasks/debian_release_repos.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+- name: 'Install Foreman GPG key'
+  apt_key:
+    url: https://deb.theforeman.org/foreman.asc
```
Comment (evgeni) [inline playbooks/roles/foreman_repositories/tasks/debian_release_repos.yml]: Using an HTTPS url here requires `ca-certificates` to be installed. this is the case on the default vagrant debian box, but may be not on others. May I suggest adding the following?

```yaml
- name: install ca-certificates
  apt:
    name: ca-certificates
    state: present
```

before this task?

**Comment by evgeni** [comment]
as mentioned on irc this (after your rebase and changes) now worked fine on my Debian laptop. Will try the other one where it was failing before later. I guess the diff is that this box runs in en_US and the failing one was en_GB.

**Comment by evgeni** [comment]
Confirmed, when I use my Fedora box with all `LC_*` vars set to en_US.UTF-8, it works fine too.

**Comment by evgeni** [comment]
Finally!

```patch
diff --git a/playbooks/roles/foreman_installer/tasks/locales.yml b/playbooks/roles/foreman_installer/tasks/locales.yml
index dbbb2a1..427beb1 100644
--- a/playbooks/roles/foreman_installer/tasks/locales.yml
+++ b/playbooks/roles/foreman_installer/tasks/locales.yml
@@ -7,3 +7,11 @@
 - name: 'Ensure en_US.UTF-8 locale is available'
   locale_gen:
     name: en_US.UTF-8
+
+- name: configure /etc/default/locale
+  copy:
+    dest: /etc/default/locale
+    content: |
+      LC_ALL="en_US.UTF-8"
+      LANG="en_US.UTF-8"
+      LANGUAGE="en_US.UTF-8"
```

after that I can deploy the box from my en_GB environment.

File: playbooks/roles/foreman_repositories/tasks/main.yml [hunk 3]
```diff
@@ -1,6 +1,9 @@
 ---
-- include: release_repos.yml
-  when: foreman_repositories_use_release and not foreman_repositories_use_koji
+- include: debian_release_repos.yml
```
Comment (evgeni) [inline playbooks/roles/foreman_repositories/tasks/main.yml]: How about `include: "{{ ansible_os_family }}_release_repos.yml"`? allows us to fold both families into one include line with fewer `when` statements.

Comment (ekohl) [inline playbooks/roles/foreman_repositories/tasks/main.yml]: Is this a common pattern? I'm not that familiar with ansible. @ehelms?

**Comment by ekohl** [comment]
> after that I can deploy the box from my en_GB environment.

Do you use a box different than the standard debian vagrant image?

**Comment by evgeni** [comment]
No, default Debian Vagrant image (https://atlas.hashicorp.com/debian/boxes/jessie64).
However, my environment is `en_US.UTF-8` with `LC_MONETARY`, `LC_NUMERIC` and `LC_MEASUREMENT` set to `en_GB.UTF-8`.

File: playbooks/roles/foreman_repositories/tasks/main.yml (hunk 3 above)
Comment (evgeni) [inline playbooks/roles/foreman_repositories/tasks/main.yml]: I like it, as it removes quite some "skipped" lines (depending on the Ansible version) from the output.

**Comment by ekohl** [comment]
Rebased and added the default locales. I haven't added puppet 5 support yet, but that shouldn't be too hard.

**Comment by ekohl** [comment]
@evgeni could you re-review?

**Comment by evgeni** [comment]
This seems to work fine for basic Foreman for me (I did not test proxy and Katello will be obviously broken).

One thing though: there are no Stretch repos on apt.puppet.com yet (see https://tickets.puppetlabs.com/browse/PA-515), but Foreman installs fine with Puppet 4.8 from Stretch itself, so I had applied the following patch:
```diff
diff --git a/roles/puppet_repositories/tasks/puppetlabs-4-debian.yml b/roles/puppet_repositories/tasks/puppetlabs-4-debian.yml
index 0f656ce..f44001a 100644
--- a/roles/puppet_repositories/tasks/puppetlabs-4-debian.yml
+++ b/roles/puppet_repositories/tasks/puppetlabs-4-debian.yml
@@ -2,8 +2,10 @@
 - name: 'Install Puppetlabs GPG key'
   apt_key:
     url: http://apt.puppetlabs.com/DEB-GPG-KEY-puppet
+  when: ansible_distribution_release != 'stretch'
 
 - name: 'Install Puppetlabs PC1 repository'
   apt_repository:
     repo: deb http://apt.puppetlabs.com {{ ansible_distribution_release }} PC1
     state: present
+  when: ansible_distribution_release != 'stretch'
```

otherwise ACK and let's get that merged

**Comment by ekohl** [comment]
Rebased & updated for Stretch.

**Comment by evgeni** [comment]
ACK

File: roles/update_os_packages/tasks/main.yml [hunk 4]
```diff
@@ -1,8 +1,17 @@
 ---
-- name: 'Update packages'
+- name: 'RedHat | Update packages'
   yum:
     name: '*'
     update_cache: yes
     state: latest
   tags:
     - packages
+  when: ansible_os_family == 'Redhat'
```
Comment (ehelms) [inline roles/update_os_packages/tasks/main.yml]: Does the `h` need to be capitalized?

Comment (ekohl) [inline roles/update_os_packages/tasks/main.yml]: Good catch.

**Comment by ekohl** [comment]
Thanks all!

**Pull request merged**